### PR TITLE
Display reference numbers on payment failure page

### PR DESCRIPTION
**Pull request by kingarj**
## Checklist:

- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes the link to an issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/payments_controller.rb [hunk 1]
```diff
@@ -46,6 +46,9 @@ def index
   def create
     payment = Payment.new(session[:vehicle_details], payments_url)
     SessionManipulation::SetPaymentId.call(session: session, payment_id: payment.payment_id)
+    SessionManipulation::SetPaymentDetails.call(session: session, email: nil,
+                                                payment_reference: payment.payment_reference,
+                                                external_id: payment.external_id)
```
Comment (miqs1992) [inline app/controllers/payments_controller.rb]: I don't understand this change. We display the same information on the success page. Why can't we use the same approach? In PaymentsController.index we can move `save_payment_details(payment)` before the if statement and use same serialization we use in PaymentsController.success

Comment (kingarj) [reply app/controllers/payments_controller.rb]: @miqs1992 good point - I've updated it to reflect this.

File: app/models/payment.rb [hunk 2]
```diff
@@ -30,6 +30,14 @@ def payment_id
...
   end
 
+  def payment_reference
+    payment_details['centralReference']
+  end
+
+  def external_id
+    payment_details['paymentProviderId']
+  end
+
```
Comment (miqs1992) [inline app/models/payment.rb]: can you add tests for this in spec/models/payment_spec.rb ?

File: spec/requests/payments_controller/index_spec.rb [hunk 3]
```diff
@@ -50,15 +50,15 @@
...
       end
 
       it 'doesn\'t set payment reference in the session' do
-        expect(session[:vehicle_details]['payment_reference']).to be_nil
+        expect(session[:vehicle_details]['payment_reference']).to eq(payment_reference)
       end
 
       it 'doesn\'t set external id in the session' do
-        expect(session[:vehicle_details]['external_id']).to be_nil
+        expect(session[:vehicle_details]['external_id']).to eq(external_id)
```
Comment (miqs1992) [inline spec/requests/payments_controller/index_spec.rb]: Fix test description - now it DOES set

File: app/controllers/payments_controller.rb [hunk 4]
```diff
@@ -77,10 +77,12 @@ def success
   #    GET /payments/failure
   #
   # ==== Params
-  # * +payment_id+ - vehicle registration number, required in the session
+  # * +payment_reference+ - payment reference, required in the session
+  # * +external_id+ - external payment id, required in the session
   #
   def failure
     @payment_reference = vehicle_details('payment_reference')
+    @external_id = vehicle_details('external_id')
```
Comment (miqs1992) [inline app/controllers/payments_controller.rb]: You can use `PaymentDetails` model here

**Comment by cruddasj** [comment]
Hi both - if this has now been resolved please can we run a re-review?

**Pull request merged**